src/sentry/tasks/integrations/github/pr_comment.py

```diff
@@ -234,7 +238,60 @@ def github_comment_workflow(pullrequest_id: int, project_id: int):
         cache.delete(cache_key)
 
         if ISSUE_LOCKED_ERROR_MESSAGE in e.json.get("message", ""):
+            metrics.incr("github_pr_comment.issue_locked_error")
+            return
+
+        elif RATE_LIMITED_MESSAGE in e.json.get("message", ""):
+            metrics.incr("github_pr_comment.rate_limited_error")
             return
 
         metrics.incr("github_pr_comment.api_error")
         raise e
+
+
+@instrumented_task(name="sentry.tasks.integrations.github_comment_reactions")
+def github_comment_reactions():
+    logger.info("github.pr_comment.reactions_task")
+
+    comments = PullRequestComment.objects.filter(
+        created_at__gte=datetime.now(tz=timezone.utc) - timedelta(days=30)
+    ).select_related("pull_request")
+
+    for comment in RangeQuerySetWrapper(comments):
+        pr = comment.pull_request
+        try:
+            repo = Repository.objects.get(id=pr.repository_id)
+        except Repository.DoesNotExist:
+            metrics.incr("github_pr_comment.comment_reactions.missing_repo")
+            continue
+
+        integration = integration_service.get_integration(integration_id=repo.integration_id)
+        if not integration:
+            logger.error(
+                "github.pr_comment.comment_reactions.integration_missing",
+                extra={"organization_id": pr.organization_id},
+            )
+            metrics.incr("github_pr_comment.comment_reactions.missing_integration")
+            return
+
+        installation = integration_service.get_installation(
+            integration=integration, organization_id=pr.organization_id
+        )
+
+        # GitHubAppsClient (GithubClientMixin)
+        # TODO(cathy): create helper function to fetch client for repo
+        client = installation.get_client()
+
+        try:
+            reactions = client.get_comment_reactions(repo=repo.name, comment_id=comment.external_id)
+
+            comment.reactions = reactions
+            comment.save()
+        except ApiError as e:
+            if RATE_LIMITED_MESSAGE in e.json.get("message", ""):
+                metrics.incr("github_pr_comment.comment_reactions.rate_limited_error")
```

Comment:
Let's quit the entire task if we hit the rate limit.

Reply:
If you do get rate limited, should this task be rescheduled for later when the rate limit has ended?

Reply:
It's not a critical task and we don't lose data by waiting for the next execution time. Ending it for the day should be fine.